packages/blade/src/components/DatePicker/Calendar.web.tsx

```diff
@@ -0,0 +1,178 @@
+  };
+
+  const handlePreviousMonth = () => {
+    const nextDate = dayjs(currentDate).subtract(columnsToScroll, 'month').toDate();
+    onPreviousMonth?.(nextDate);
+    onPrevious?.(nextDate);
+    setDate(nextDate);
+  };
+
+  const handleNextYear = () => {
+    const nextDate = dayjs(currentDate).add(columnsToScroll, 'year').toDate();
+    onNextYear?.(nextDate);
+    onNext?.(nextDate);
+    setDate(nextDate);
+  };
+
+  const handlePreviousYear = () => {
+    const nextDate = dayjs(currentDate).subtract(columnsToScroll, 'year').toDate();
+    onPreviousYear?.(nextDate);
+    onPrevious?.(nextDate);
+    setDate(nextDate);
+  };
+
+  const handleNextDecade = () => {
+    const nextDate = dayjs(currentDate)
+      .add(10 * columnsToScroll, 'year')
+      .toDate();
+    onNextDecade?.(nextDate);
+    onNext?.(nextDate);
+    setDate(nextDate);
+  };
+
+  const handlePreviousDecade = () => {
+    const nextDate = dayjs(currentDate)
+      .subtract(10 * columnsToScroll, 'year')
+      .toDate();
+    onPreviousDecade?.(nextDate);
+    onNext?.(nextDate);
+    setDate(nextDate);
+  };
+
+  return (
+    <CalendarStyles
+      display="flex"
+      flexDirection="column"
+      gap="spacing.7"
+      pickerType={levelToPicker[level]}
+    >
+      <CalendarHeader
+        isRange={isRange}
+        date={currentDate}
+        onLevelChange={(level) => setLevel(() => level)}
+        pickerType={levelToPicker[level] as PickerType}
+        onNextMonth={handleNextMonth}
+        onPreviousMonth={handlePreviousMonth}
+        onNextDecade={handleNextDecade}
+        onPreviousDecade={handlePreviousDecade}
+        onNextYear={handleNextYear}
+        onPreviousYear={handlePreviousYear}
+      />
+      <CalendarGradientStyles isRange={isRange} date={currentDate}>
+        <DatePicker
+          withCellSpacing={false}
+          type={isRange ? 'range' : 'default'}
+          date={_date}
+          locale={locale}
+          level={level}
+          onDateChange={setDate}
+          onLevelChange={(level) => setLevel(() => level)}
+          numberOfColumns={numberOfColumns}
+          weekdayFormat="ddd"
+          firstDayOfWeek={firstDayOfWeek}
+          // @ts-expect-error unable to narrow props based on `type`
+          allowSingleDateInRange={allowSingleDateInRange}
+          renderDay={(date) => {
+            return (
+              <BaseBox data-date={`${date.getMonth()}-${date.getDate()}`}>{date.getDate()}</BaseBox>
+            );
+          }}
+          classNames={{
+            levelsGroup: 'DatePicker-levelsGroup',
+            day: 'DatePicker-cell',
+            monthsListControl: 'DatePicker-cell',
+            yearsListControl: 'DatePicker-cell',
+            calendarHeader: 'DatePicker-header',
+            monthRow: 'DatePicker-row',
+            yearsListRow: 'DatePicker-row',
+            monthsListRow: 'DatePicker-row',
+            weekdaysRow: 'DatePicker-row',
+            weekday: 'DatePicker-weekday',
```

Comment:
maintain this in some `tokens.ts / constants.ts` file maybe? since we're reusing these strings while writing CSS also

Reply:
done.